app/src/app/components/routes/Router.tsx

```diff
@@ -7,21 +7,25 @@ import {NotFound} from "../NotFound";
 import {Download} from "../contents/download";
 import {ChangeLogs} from "../contents/changelogs";
 import {Metadata} from "../contents/metadata";
+import {Login, Redirect} from "../login";
+import {ProtectedRoute} from "./ProtectedRoute";
 
 export function Router() {
     return (
         <Routes>
-            <Route path="*" element={<NotFound/>}/>
+            <Route path={"login"} element={<Login/>}/>
+            <Route path="redirect" element={<Redirect/>}/>
+            <Route path="*" element={<ProtectedRoute element={<NotFound/>}/>}/>
             <Route element={<Sidebar/>}>
-                <Route index element={<Explorer/>}/>
-                <Route path="packets" element={<Explorer/>}/>
-                <Route path="run" element={<PacketRunner/>}/>
-                <Route path="run-workflow" element={<WorkflowRunner/>}/>
-                <Route path="documentation" element={<ProjectDocumentation/>}/>
-                <Route path="packets/:packetId" element={<PacketDetails/>}/>
-                <Route path="packets/:packetId/metadata" element={<Metadata/>}/>
-                <Route path="packets/:packetId/downloads" element={<Download/>}/>
-                <Route path="packets/:packetId/changelogs" element={<ChangeLogs/>}/>
+                <Route index element={<ProtectedRoute element={<Explorer/>}/>}/>
+                <Route path="packets" element={<ProtectedRoute element={<Explorer/>}/>}/>
+                <Route path="run" element={<ProtectedRoute element={<PacketRunner/>}/>}/>
+                <Route path="run-workflow" element={<ProtectedRoute element={<WorkflowRunner/>}/>}/>
+                <Route path="documentation" element={<ProtectedRoute element={<ProjectDocumentation/>}/>}/>
+                <Route path="packets/:packetId" element={<ProtectedRoute element={<PacketDetails/>}/>}/>
+                <Route path="packets/:packetId/metadata" element={<ProtectedRoute element={<Metadata/>}/>}/>
+                <Route path="packets/:packetId/downloads" element={<ProtectedRoute element={<Download/>}/>}/>
+                <Route path="packets/:packetId/changelogs" element={<ProtectedRoute element={<ChangeLogs/>}/>}/>
```

Comment:
Since you are just doing protected route in all of these, you could just use a route element that wraps all of these and just have the protected route logic in there with an `<Outlet/>` if they are authenticated?

Reply:
Done